server/http.go

```diff
@@ -0,0 +1,66 @@
+package server
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/skuid/keyman/oidcauth"
+	"github.com/skuid/keyman/shapes"
+	"go.uber.org/zap"
+)
+
+// SignHTTP is a handler for incoming certificate requests. Requests must contain
+// an oidcauth.User struct in the context
+func (a *Authority) SignHTTP(w http.ResponseWriter, r *http.Request) {
+	var (
+		input    = &shapes.SignRequest{}
+		response = &shapes.KeyResponse{}
+	)
+	zap.L().Info("handling SignHTTP")
+
+	user, ok := oidcauth.FromContext(r.Context())
+	if !ok {
+		zap.L().Error("Couldn't get user from context: please wrap with oidcauth.OidcUserContext()")
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
```

Comment:
I think `http. StatusInternalServerError` makes more sense for this return, since this would be a programmer error, right?

Reply:
Yea, good call